### PR TITLE
Removed HttpNoContentFormatter from MvcOptionsSetup

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue - #2409 
@rynowak

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.FunctionalTests/ActionResultTests.cs [hunk 1]
```diff
@@ -301,7 +301,7 @@ public class ActionResultTests
 
             // Assert
             Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
-            Assert.Equal("", await response.Content.ReadAsStringAsync());
+            Assert.Equal("null", await response.Content.ReadAsStringAsync());
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/ActionResultTests.cs]: I'm not sure this is right - how is "null" getting in there?

File: test/Microsoft.AspNet.Mvc.FunctionalTests/ModelBindingTest.cs [hunk 2]
```diff
@@ -2159,7 +2159,7 @@ public async Task
             // Assert
             Assert.Equal(HttpStatusCode.NoContent, response.StatusCode);
             var body = await response.Content.ReadAsStringAsync();
-            Assert.Equal(string.Empty, body);
+            Assert.Equal("null", body);
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/ModelBindingTest.cs]: seems suspect.

Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/ModelBindingTest.cs]: Discussed in person, the action code is specifically creating an `ObjectResult() { Value = null }`

File: test/Microsoft.AspNet.Mvc.FunctionalTests/ActionResultTests.cs (hunk 1 above)
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/ActionResultTests.cs]: Discussed in person, the action code is specifically creating an `ObjectResult() { Value = null }`

**Comment by rynowak** [comment]
:shipit:

File: test/Microsoft.AspNet.Mvc.FunctionalTests/ActionResultTests.cs (hunk 1 above)
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/ActionResultTests.cs]: Discussed in person, the action code is specifically creating an `ObjectResult() { Value = null }`

**Comment by rynowak** [comment]
I think I jumped the gun on this, we want to change the behavior for `void` but not for returning a `null`. Let's make the change in the invoker

**Comment by ajaybhargavb** [comment]
Created a new PR with the invoker changes. https://github.com/aspnet/Mvc/pull/2687
Closing this one.

**Pull request closed without merging**